File: rust/src/insertion_sort.rs
fn insertion_sort(arr: &mut Vec<i32>) -> &mut Vec<i32> {
  for i in 1..arr.len() - 1 {
      let mut j = i - 1;
      let value = arr[i];
      while j > 0 {
          if arr[i] < arr[j] {
              arr.swap(i, j);
              j -= 1;
          } else {
              break
          }
      }
  }
  arr
}

File: typescript/sort/insertionSort.ts
export function insertionSort(arr: number[]) {
	for (let i = 1; i < arr.length; i++) {
		const value = arr[i];
		let j = i - 1;
		while (j >= 0) {
			if (value < arr[j]) {
				arr[j + 1] = arr[j];
				arr[j] = value;
				j--;
			} else {
				break;
			}
		}
	}
	return arr;
}
